src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php

```diff
@@ -127,19 +127,67 @@ public function getMaxDepth(): ?int
     /**
      * {@inheritdoc}
      */
-    public function setSerializedName(string $serializedName = null)
+    public function setSerializedNames(array $serializedNames): void
     {
-        $this->serializedName = $serializedName;
+        $this->serializedName = $serializedNames;
+    }
+
+    /**
+     * Set a serialization name for given groups.
+     *
+     * @param string[] $groups
+     */
+    public function setSerializedName(string $serializedName = null/* , array $groups = [] */)
+    {
+        if (\func_num_args() < 2) {
+            $groups = [];
+        } else {
+            $groups = func_get_arg(1);
+
+            if (!\is_array($groups)) {
+                throw new \TypeError(sprintf('Argument 2 passed to "%s()" must be array, "%s" given.', __METHOD__, get_debug_type($groups)));
+            }
+        }
+
+        foreach ($groups ?: ['*'] as $group) {
+            $this->serializedName[$group] = $serializedName;
+        }
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getSerializedName(): ?string
+    public function getSerializedNames(): array
     {
         return $this->serializedName;
     }
 
+    /**
+     * Gets the serialization name for given groups.
+     *
+     * @param string[] $groups
+     */
+    public function getSerializedName(/* array $groups = [] */): ?string
```

Comment:
Why pass an array here? Shouldn't we accept string|null instead?

Reply:
Used by `MetadataAwareNameConverter`
https://github.com/symfony/symfony/pull/46432/files#diff-87ffe277c0213fca753a8e0b26cc8f4436e255b9776e5e08852c403df247c5b9R121

Serializer group context can be multiple. So returned `SerializerName` is the first matching group list.

Should be done in `MetadataAwareNameConverter` ?